docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/irsa.md

```diff
@@ -0,0 +1,327 @@
+
+There are different ways to configure the mapping within OpenSearch:
+
+- Via a [Terraform module](https://registry.terraform.io/modules/idealo/opensearch/aws/latest) in case your OpenSearch instance is exposed.
+- Via the [OpenSearch dashboard](https://opensearch.org/docs/latest/security/access-control/users-roles/).
+- Via the REST API.
+
+The important part is assigning the `iam_role_arn` of the previously created `opensearch_role` to an internal role within OpenSearch. For example, `all_access` on the OpenSearch side is a good candidate, or if required, extra roles can be created with more restrictive access.
+
+### Operate
+
+Configure Operate to use the feature set of IRSA for the OpenSearch Exporter. See the configuration reference [here](../../../../operate-deployment/operate-configuration.md#elasticsearch-or-opensearch).
+
+As an example, configure the following environment variables:
+
+```
+- name: CAMUNDA_OPERATE_ELASTICSEARCH_URL
+  value: https://test-domain.region.es.amazonaws.com
+- name: CAMUNDA_OPERATE_ZEEBEELASTICSEARCH_URL
+  value: https://test-domain.region.es.amazonaws.com
+```
+
+Where the value is whatever the endpoint of your OpenSearch cluster is.
+
+:::note
+AWS OpenSearch runs generally on port 443.
+:::
+
+### Zeebe
+
+Configure Zeebe to use the feature set of IRSA for the OpenSearch Exporter. See the configuration reference [here](../../../../zeebe-deployment/configuration/broker.md#zeebebrokerexportersopensearch-opensearch-exporter).
+
+As an example, configure the following environment variables:
+
+```
+- name: ZEEBE_BROKER_EXPORTERS_OPENSEARCH_ARGS_AWS_ENABLED
+  value: true
+- name: ZEEBE_BROKER_EXPORTERS_OPENSEARCH_CLASSNAME
+  value: io.camunda.zeebe.exporter.opensearch.OpensearchExporter
+- name: ZEEBE_BROKER_EXPORTERS_OPENSEARCH_ARGS_URL
+  value: https://test-domain.region.es.amazonaws.com
+```
+
+## Troubleshooting
+
+### Instance Metadata Service (IMDS)
+
+[IMDS](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html) is a default fallback for the AWS SDK. Within the context of EKS, it means a pod will automatically assume the role of a node. This can hide many problems, including whether IRSA was set up correctly or not, since it will fall back to IMDS in case of failure and hide the actual error.
+
+Thus, if nothing within your cluster relies on the implicit node role, we recommend disabling it by defining in Terraform the `http_put_response_hop_limit`, for example.
+
+Using a Terraform module like the [EKS module](https://registry.terraform.io/modules/terraform-aws-modules/eks/aws/latest), one can define the following to decrease the default value of two to one, which results in pods not being allowed to assume the role of the node anymore.
+
+```
```

Comment:
Add `json` for consistency with other HCL sections.

Reply:
ah thanks, was wondering what those are using, awesome!

Reply:
In the end, HCL is as ugly as JSON 😂